diff --git a/tools/gnatcov/Makefile b/tools/gnatcov/Makefile
index 4de1b0f9..d1ba851a 100644
--- tools/gnatcov/Makefile
+++ tools/gnatcov/Makefile
@@ -346,7 +346,7 @@ install-adapters: $(INSTALL_ADAPTER_TARGETS)
 # which has its own extra info embedded and eventually installs just all what
 # the distrib tree contains. We don't want any toplevel item there.
 
-.PHONY: distrib-linux-extra distrib-windows-extra distrib
+.PHONY: distrib-linux-extra distrib-windows-extra distrib-darwin-extra distrib
 
 distrib-linux-extra:
 	[ -f COPYING3 ] || $(CP) ../../COPYING3 .
@@ -358,6 +358,8 @@ distrib-linux-extra:
 
 distrib-windows-extra:
 
+distrib-darwin-extra:
+
 distrib-common: install-adapters distrib-${HOST_OS}-extra
 distrib-without-doc: distrib-common install-without-doc
 distrib-with-doc: distrib-common install-with-doc
diff --git a/tools/gnatcov/host.mk b/tools/gnatcov/host.mk
index 53ba4b0b..44509578 100644
--- tools/gnatcov/host.mk
+++ tools/gnatcov/host.mk
@@ -14,6 +14,11 @@ HOST_OS=linux
 LN_S=ln -sf
 exeext=
 endif
+ifneq (,$(filter Darwin, $(HOST_UNAME)))
+HOST_OS=darwin
+LN_S=ln -sf
+exeext=
+endif
 
 RM=rm -f
 CP=cp -pf
